code/datums/elements/deferred_aquarium_content.dm

```diff
@@ -0,0 +1,25 @@
+/**
+ * Create /datum/component/aquarium_content with the preset path on the target right before being inserted into aquarium and deletes itself.
+ * Used to save memory from aquarium properties on common objects/stacks that won't see aquarium in 99 out of 100 rounds.
+ */
+/datum/element/deferred_aquarium_content
+	element_flags = ELEMENT_BESPOKE
+	id_arg_index = 2
+	var/aquarium_content_type
+
+/datum/element/deferred_aquarium_content/Attach(datum/target, aquarium_content_type)
+	. = ..()
+	if(!ismovable(target) || !aquarium_content_type)
```

Comment:
if aquarium content type isn't set it shouldn't return element_incompatible; as the element is compatible, its just got no var set (So should yell about that instead)